framework/src/setroubleshoot/server.py

```diff
@@ -698,10 +705,30 @@ def goodbye(database):
     audit2why.finish()
 
 def RunFaultServer(timeout=10):
-    # FIXME
-    audit2why.init()
+    signal.alarm(timeout)
+    sigalrm_handler = signal.signal(signal.SIGALRM, polling_failled_handler)
+    # polling for /sys/fs/selinux/policy file
+    while True:
+        try:
+            audit2why.init()
+            signal.alarm(0)
+            break
+        # retry if init() failed to open /sys/fs/selinux/policy
+        except ValueError as e:
+            # occurs in python 2
+            if "Device or resource busy" in str(e):
+                continue
+            raise e
+        except SystemError as e:
+            # in python 3 SystemError occurs as a direct cause of ValueError
+            # Error message of the ValueError is stored in __context__ (not present in python 2)
+            if "Device or resource busy" in str(getattr(e, "__context__", "")):
+                continue
+            raise e
```

Comment:
Given that we have stable - python2 - and master - python3 - branches, it's probably not necessary to mix python2 and python3 code.

Reply:
Ok, I'll rewrite it.